packages/teams-js/src/public/clipboard.ts

```diff
@@ -0,0 +1,66 @@
+import { sendAndHandleSdkError } from '../internal/communication';
+import { ensureInitialized } from '../internal/internalAPIs';
+import { errorNotSupportedOnPlatform, FrameContexts } from './constants';
+import { runtime } from './runtime';
+
+/**
+ * Namespace to interact with the clipboard specific part of the SDK.
+ *
+ * @beta
+ */
+export namespace clipboard {
+  /**
+   * Function to copy data to clipboard.
+   * @remarks
+   * Note: clipboard.write only supports Text, HTML, PNG, JPEG and SVG data format.
+   *       MIME type for Text -> `text/plain`, HTML -> `text/html`, PNG/JPEG/SVG -> `image/(png | jpeg | svg+xml)`
+   *       Also, JPEG and SVG will be converted to PNG image when copying to clipboard.
+   *
+   * @param blob - A Blob object representing the data to be copied to clipboard.
+   * @returns A string promise which resolves to success message from the clipboard or
+   *          rejects with error stating the reason for failure.
+   */
+  export function write(blob: Blob): Promise<string> {
+    ensureInitialized(runtime, FrameContexts.content, FrameContexts.task, FrameContexts.stage, FrameContexts.sidePanel);
+    if (!isSupported()) {
+      throw errorNotSupportedOnPlatform;
+    }
+    if (
+      (blob.type.startsWith('image') &&
+        !blob.type.endsWith('png') &&
+        !blob.type.endsWith('jpeg') &&
+        !blob.type.endsWith('svg+xml')) ||
+      (blob.type.startsWith('text') && !blob.type.endsWith('plain') && blob.type.endsWith('html'))
+    ) {
+      throw new Error(`Blob type ${blob.type} is not supported.`);
+    }
+    return sendAndHandleSdkError('clipboard.writeToClipboard', blob);
+  }
+
+  /**
+   * Function to read data from clipboard.
+   *
+   * @returns A promise blob which resolves to the data read from the clipboard or
+   *          rejects stating the reason for failure.
+   *          Note: Returned blob type will contain one of the MIME type `image/png`, `text/plain` or `text/html`.
+   */
+  export function read(): Promise<Blob> {
+    ensureInitialized(runtime, FrameContexts.content, FrameContexts.task, FrameContexts.stage, FrameContexts.sidePanel);
```

Comment:
How was this list of FrameContexts chosen? Nothing in here seems wrong to me, but curious why ones like "authentication" for example don't allow it.

Reply:
Considering security issues, we want to avoid developers access to end user's password or any misuse of the API.

Reply:
i never knew this was an issue!  i learned something :)